### DISCUSSION TITLE
Saving a group of projects

### DISCUSSION THREAD

**Discussion by mambax7** [WinMerge/winmerge#1695] [Ideas]
A lot of times, I am comparing several different folders in many tabs. I can save each tab as a project, and then load it. 

But it would be awesome if I could save **ALL** currently opened projects in all tabs as a "group", and then load the whole group at once.

**Comment by sdottaka** [comment]
There is currently no GUI support, but you can include multiple sets of paths in your project file by editing it manually:
~~~
<?xml version="1.0" encoding="UTF-8"?>
<project>
	<paths>
		<left>c:\dev\winmerge\src</left>
		<right>d:\dev\winmerge\src</right>
...
	</paths>
	<paths>
		<left>c:\dev\winmerge\plugins</left>
		<right>d:\dev\winmerge\plugins</right>
...
	</paths>
</project>
~~~

**Comment by mambax7** [comment]
Thank you so much for the tip, I was not aware of it. 
But if it would be possible to add it one day in the future in GUI the option to save a group, it would be fantastic. 
This way I wouldn't have to do anything manually! :)

**Comment by mcoulont** [comment]
Hello

Is the project file editing solution supposed to work with WinMerge_2.16.16 (portable version)?

In my case verything goes as though only the first `paths` tag was present.

**Reply by sdottaka** [reply]
WinMerge 2.16.3 and later supports multiple `<paths>` in a project file.
I tried opening the following project in WinMerge 2.16.16, and multiple tabs opened:
~~~
<?xml version="1.0" encoding="UTF-8"?>
<project>
	<paths>
		<left>Compare\Dir1\</left>
		<right>Compare\Dir2\</right>
		<filter>*.*</filter>
		<subfolders>1</subfolders>
		<left-readonly>0</left-readonly>
		<right-readonly>0</right-readonly>
	</paths>
	<paths>
		<left>Compare\Dir1\</left>
		<right>Compare\Dir2\</right>
		<filter>*.*</filter>
		<subfolders>1</subfolders>
		<left-readonly>0</left-readonly>
		<right-readonly>0</right-readonly>
	</paths>
	<paths>
		<left>Compare\Dir1\</left>
		<right>Compare\Dir2\</right>
		<filter>*.*</filter>
		<subfolders>1</subfolders>
		<left-readonly>0</left-readonly>
		<right-readonly>0</right-readonly>
	</paths>
</project>

~~~

**Reply by mcoulont** [reply]
> WinMerge 2.16.3 and later supports multiple `<paths>` in a project file. I tried opening the following project in WinMerge 2.16.16, and multiple tabs opened:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <project>
> 	<paths>
> 		<left>Compare\Dir1\</left>
> 		<right>Compare\Dir2\</right>
> 		<filter>*.*</filter>
> 		<subfolders>1</subfolders>
> 		<left-readonly>0</left-readonly>
> 		<right-readonly>0</right-readonly>
> 	</paths>
> 	<paths>
> 		<left>Compare\Dir1\</left>
> 		<right>Compare\Dir2\</right>
> 		<filter>*.*</filter>
> 		<subfolders>1</subfolders>
> 		<left-readonly>0</left-readonly>
> 		<right-readonly>0</right-readonly>
> 	</paths>
> 	<paths>
> 		<left>Compare\Dir1\</left>
> 		<right>Compare\Dir2\</right>
> 		<filter>*.*</filter>
> 		<subfolders>1</subfolders>
> 		<left-readonly>0</left-readonly>
> 		<right-readonly>0</right-readonly>
> 	</paths>
> </project>
> ```

OK it works with your paths.

Except that when I go a tab, only the differing lines appear. But the problem is solved by correcting at `Display->Context diff`.
